apps/omg_watcher/lib/exit_processor/core.ex

```diff
@@ -485,7 +668,43 @@ defmodule OMG.Watcher.ExitProcessor.Core do
          do: double_spent_input
   end
 
+  # this function doesn't care, if the second argument holds additional information about the utxo position
+  defp competitor_for(signed1, %KnownTx{signed_tx: signed2}),
+    do: competitor_for(signed1, signed2)
+
+  # it also doesn't care if the second argument is signed or not
+  defp competitor_for(signed1, %Transaction.Signed{raw_tx: known_raw_tx}),
+    do: competitor_for(signed1, known_raw_tx)
+
   defp get_known_txs(%__MODULE__{} = state) do
     TxAppendix.get_all(state)
+    |> Enum.map(fn signed -> %KnownTx{signed_tx: signed} end)
+  end
+
+  defp get_known_txs(%Block{transactions: txs, number: blknum}) do
+    txs
+    |> Enum.map(fn tx_bytes ->
+      %Transaction.Recovered{signed_tx: signed} = recover_correct_tx_from_block(tx_bytes)
+      signed
+    end)
+    |> Enum.with_index()
+    |> Enum.map(fn {signed, txindex} -> %KnownTx{signed_tx: signed, utxo_pos: Utxo.position(blknum, txindex, 0)} end)
+  end
+
+  defp get_known_txs([]), do: []
+  defp get_known_txs([%Block{} | _] = blocks), do: blocks |> Enum.flat_map(&get_known_txs/1)
+
+  defp recover_correct_tx_from_block(tx_bytes) do
+    {:ok, recovered} = OMG.API.Core.recover_tx(tx_bytes)
+    recovered
+  end
+
+  # based on an enumberable of `Utxo.Position` and a mapping that tells whether one exists it will pick
+  # only those that **were checked** and were missing
+  # (i.e. those not checked are assumed to be present)
+  defp only_utxos_checked_and_missing(utxo_positions, utxo_exists?) do
+    # the default value below is true, so that the assumption is that utxo not checked is **present**
+    utxo_positions
+    |> Enum.filter(fn pos -> !Map.get(utxo_exists?, pos, true) end)
```

Comment:
This one can be more effectively done using `MapSet.difference/2`.

Reply:
I'll leave a TODO, but as discussed, there are much more expensive calls around here